### DISCUSSION TITLE
'주문에 필요한 정보'에 대한 정리 (헷갈리는 용어에 대한 사항)

### DISCUSSION THREAD

**Discussion by Topslug** [cau-swe/checkpoint-1#45] [General]
주문할 때 필요할 만한 정보들(배달/pick-up에 필요한 정보)
- 주문한 피자(1개이상 가능) 각각의 수량
- 배달인지 pick-up인지
   - 배달일 경우 주문자 위치
- 본인 확인 용 주문자 번호
- 추가 요청 사항(직원에게 하고 싶은 말)

이외에 더 필요한 게 있다고 생각하시는 분은 의견을 남겨주시면 될 것 같습니다!
그리고 저희끼리 용어를 통일 해야 될 필요성을 느끼고 있는데
저는 여기에 적힌 정보들을 통째로 '주문 정보'라고 약속하는 것이 어떨까 싶습니다.
이견이 있으신 분들은 공유해주시면 좋을 것 같습니다.

추가로 Employee팀에서 '주문자 정보'로 주문자 번호, 주문자 위치를 말씀하셨습니다.

앞으로 의사소통에 있어서 헷갈리는 사항(용어)들은 서로 통일을 하면 좋을 것 같습니다.

추가로 제가 생각했던 직원 정보와 직원 계정 정보를 여기서 공유하도록 하겠습니다.
그리고 현재 Customer팀에서 피자에 대한 상세 정보를 취합하고 있는데 우선 여기다가 Administrator팀에서 생각했던 피자 관리 시 필요한 정보와 고객 정보에 대해서 공유해 놓도록 하겠습니다.

주문 기록에 대해서 어떤 정보를 기록하면 좋을까요?
주문관리페이지(현재 주문 목록)에서는 어떤 정보들이 display될 수 있을까요?(제 생각에는 주문 정보에 가격까지 포함되면 될 것 같은데 맞나요? 아니면 주문 정보에 가격 정보를 포함할까요? 어쨌든 시스템에서 총 가격 계산은 해줘야 될 것 같은데, 주문 기록 정보로 따로 관리할지, 아니면 주문 정보에 가격을 포함시켜서 같이 관리할 지 여러분들의 의견이 궁금합니다.)

정리

1. 주문 정보

- 주문한 피자 종류(1개이상 가능) 각각의 수량
- 배달인지 pick-up인지
- ( 배달일 경우 주문자 위치 )
- 본인 확인 용 주문자 번호
- 추가 요청 사항(직원에게 하고 싶은 말)

2. 주문자 정보

- 주문자 번호
- 주문자 위치

3. 직원 정보
- 이름
- 전화번호
- 월급

4. 직원 계정 정보(관리자 계정 정보와 구성이 같음)
- ID
- 비밀번호

5. 피자 정보
- 사진
- 피자 이름
- 설명(ex 달콤한 고구마와 치즈가 어울려진 000피자)
- 가격

6. 고객 정보
- 이름
- 전화번호
- 주소
- ID
- 비밀번호

7. Review 정보
- 피자 이름
- 일시
- 내용

8. 주문 기록 정보
- 주문 정보
- 일시

9. 피자 상세 정보(고객들의 추가적인 클릭에 의해 한 번 더 보여지는 정보)
- 피자 정보 + Review 정보

**Comment by TedLeem** [comment]
피자 상세정보는 언급해주신  것에 후기를 추가하여 아래와 같이 결정하면 될 것 같습니다.   
피자 정보
> 사이즈
> 사진
> 피자 이름
> 설명(ex 달콤한 고구마와 치즈가 어울려진 000피자)
> 가격  
> 후기

**Reply by Topslug** [reply]
아 그렇군요, Review를 생각을 못했었네요, 
그런데, 하나의 피자에 여러 개의 Review가 달릴 것 같은데, 
Review정보는 피자 이름, 일시, 게시자, 리뷰 내용
이런 식으로 따로 정해야 되지 않을까요?

**Reply by TedLeem** [reply]
아직 리뷰정보를 제대로 정하지 않았는데,   
상민님이 말씀해주신 것처럼 하면 될 것 같은데요?
> 피자이름
> 일시
> 게시자
> 리뷰내용

게시자를 빼서 좀더 간소화 시켜도 좋을 것 같습니다

**Reply by Topslug** [reply]
아! 그렇군요, 게시자를 넣은 것은 나중에 고객에게 display될 때 조금 신뢰성을 확보할 수 있지 않을까 싶었는데, 우선 게시자를 제외하고 업데이트를 해놓겠습니다.

**Comment by Topslug** [comment]
개인적으로 피자 정보와 피자 상세 정보라는 말을 분리하는 게 어떤지 제안 드립니다.
피자 정보는 피자 메뉴 목록에 저장되어 있는 피자 제품에 관한 정보 = 고객이 사이트 접속 시 바로 목록에서 확인 가능한 정보
피자 상세 정보는 고객들의 추가적인 클릭에 의해 한 번 더 보여지는 정보 (피자 정보 + 해당 피자의 Review)

**Reply by TedLeem** [reply]
좋습니다!

**Comment by Topslug** [comment]
그리고 생각해보니 피자 사이즈(S, M, L)를 생각하긴 했었는데, Customer쪽에서 피자의 크기를 선택하는 경우는 없었던 것 같습니다. 프로그램의 간소화를 위해 피자 사이즈는 없애도 되지 않을까 싶은데 Customer팀의 의견이 필요할 것 같습니다.

**Reply by TedLeem** [reply]
넵 저희도 간소화를 위해 빼는게 좋다고 생각했습니다!

**Reply by Topslug** [reply]
넵! 그러면 빼도록 하겠습니다!

**Comment by Topslug** [comment]
그리고 주문 정보와 직원 화면에서 display되는 주문 정보가 구성이 같은지 다른지 얘기해보면 좋을 것 같습니다.
왜냐하면 pick-up을 하는 경우가 있기 때문에 가격이 포함돼서 display되면 좋을 것 같은데, 가격이 계산되는 시점이 헷갈리는 것 같습니다.
고객이 주문을 완료한 시점에 가격까지 주문 정보에 포함돼서 오는 것인지, 아니면 고객은 메뉴만 선택한 것이므로 매장에 도착했을 때 가격이 계산돼서 display되는 것인지 헷갈립니다.(Use Case상으로는 고려하지 않아도 될 수도 있지만)
주문 정보에 포함 시키는 것이  간단할 것 같다는 생각은 듭니다.

**Comment by Topslug** [comment]
주문 정보에서 추가 요청에 대한 사항

저는 개인적으로 추가 요청 사항을 수정했으면 좋겠습니다. (삭제해도 좋습니다.)
왜냐하면 추가 요청 사항은 고객들의 주관적인 정보로 해당 내용이 길지 짧을지 예측할 수 없기 때문입니다.
반면 옵션을 줘서 목록 중에서 선택하도록 한다면 있는지 없는지, 어떤 게 있는지, 구체적인 상황과 크기를 예측할 수 있습니다.
이러한 점에서 Employee쪽 UI에 주문 정보를 한번에 조회할 수 있는 게 가능한지, 추가적인 클릭을 통해 요청 사항을 확인해야 되는 지에 영향을 미치게 된다고 생각합니다.

그리고 비즈니스 측면에서 생각했을 때, 피자 메뉴 자체로는 뭔가 추가하고, 덜고 하는 것이 이상하게 느껴집니다.
대신 추가 소스 요청, 추가 피클 요청 -> 이 부분은 추가 결제 사항이 될 것 같아서 추가 요청 사항 대신, 사이드 메뉴 선택, 소스 선택 등이 더 적합하지 않은가 생각합니다.

해장국, 설렁탕 등에 대해서는 소금을 더 쳐서 짭잘하게 해달라든지, 김치를 좀 더 많이 담아주세요라고 요청 하는 게 가능할 것도 같은데(이런 것도 효율성을 고려해 선택 사항으로 만들 수 있을 것 같다고 생각합니다. 왜냐하면 통계적으로 봤을 때 자주 요청되는 사항이 있기 때문입니다.), 

피자같은 경우는 거의 다 단일 제품화 되어 있기 때문에 어렵지 않을까 생각합니다.(피클은 포장되서 판매되고, 소스도 팩에 담겨져 판매됩니다. 보통 피자 가게 같은 경우, 치즈 크러스트도 사장님 재량이 아닌 추가 사항으로 명시되어 있습니다.) 

그리고 (예측이 불가능한)추가 요청 사항을 고려하고 확인하는 시간이 주문 처리 시간에 영향을 미치게 된다고 생각합니다. 최대한 줄일 수 있으면 줄이는 게 간편해질 거라고 생각합니다.

따라서 저는 추가 요청 사항에 대해 옵션으로 선택하게 함으로써 UI를 더욱 간편하게 하고, UI동선을 간소화하는 게 좋을 것 같다고 생각합니다.
